src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -109,6 +109,24 @@ public Conversion ClassifyConversionForCast(BoundExpression source, TypeSymbol d
             return implicitConversion.Exists ? implicitConversion : Conversion.NoConversion;
         }
 
+        /// <summary>
+        /// returns true when implicit conversion is not necessarily the same as explicit conversion
+        /// </summary>
+        private static bool ExplicitConversionMayDifferFromImplicit(Conversion implicitConversion)
+        {
+            switch (implicitConversion.Kind)
+            {
+                case ConversionKind.ImplicitUserDefined:
```

Comment:
User defined and dynamic are special. The others just can contain them directly or indirectly. 
Could become a property of Conversion to be more precise.